### PR TITLE
Add robot and geolyzer API function signatures

### PR DESCRIPTION
**Pull request by freja-p**
Docstring explanations taken from OC online reference https://ocdoc.cil.li/component:geolyzer and https://ocdoc.cil.li/api:robot

### CODE REVIEW

File: library/lib/robot.lua
Comment (AR2000AR) [inline library/lib/robot.lua:1]: The robot library is not the component. You cannot make it a Component class

File: library/lib/robot.lua [hunk 1]
```diff
@@ -0,0 +1,323 @@
+---@meta
+
+---@class ComponentRobot : Component
```
Comment (AR2000AR) [inline library/lib/robot.lua]:
```suggestion
---@class libRobot
```

File: library/lib/robot.lua [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+---@param count? number
+---@return boolean
+function robot.fillUp(count)
+end
+
+---Like fill, but operates on the block below the robot.
+---@param count? number
+---@return boolean
+function robot.fillDown(count)
+end
```
Comment (AR2000AR) [inline library/lib/robot.lua]:
```suggestion
end
return robot
```

File: library/lib/component.lua [hunk 3]
```diff
@@ -21,6 +21,8 @@
 ---@field os_cardwriter ComponentOsCardWriter
 ---@field stargate ComponentStargate
 ---@field redstone ComponentRedstone
+---@field geolyzer ComponentGeolyzer
+---@field robot ComponentRobot
```
Comment (AR2000AR) [inline library/lib/component.lua:25]:
```suggestion

```

File: library/lib/component.lua
Comment (AR2000AR) [inline library/lib/component.lua:1]: You didn't make the definition file for the robot component, but for the robot library

**Review by AR2000AR — CHANGES_REQUESTED** [review]

**Comment by AR2000AR** [comment]
Thank you for the pr, but as it is I cannot merge it. I will once the issue I mentioned are fixed

File: library/lib/component.lua
Comment (freja-p) [reply library/lib/component.lua:1]: Ah, I was referring to the robot API doc for the functions, did not realise there was a component.robot doc which the API is built from, which is why I tried to make it into a library instead. 

I'll add the component.robot as component/robot.d.lua and the robot API as the robot.lua lib

File: library/lib/component.lua (hunk 3 above)
Comment (freja-p) [reply library/lib/component.lua:25]: I've added the proper ComponentRobot definition file to the PR

**Review by AR2000AR — APPROVED** [review]

**Pull request merged**